### ISSUE TITLE
Provision new AWS resources for API notification delivery

### ISSUE THREAD

**Issue by jwalgran** [open-apparel-registry/open-apparel-registry#1998]
- Provision an AWS Batch queue for notifications
  - Use list processing Batch jobs as a reference
  - Send batch job failure notifications using the same method as list processing failure notifications
- Provision a new AWS Batch compute environment
  - Notification delivery will require modest resources. Configure compute to have access to as many instance types as are appropriate to launch a Django container that just makes db querys and POSTs  to URLs
- ~Provision a CloudWatch log stream for logging sent notifications~ (details on this removal in comments below)
  - ~The "CLI" ECS tasks will need write permissions to the log stream~

**Comment by jwalgran** [comment]
On 2022-08-05 @maurizi  and I chatted about CloudWatch in the context of #2000.  The content CloudWatch of log groups can be filtered and parsed in powerful ways so we can use our existing log group and just filter it to get information about notifications

<img width="960" alt="Screen Shot 2022-08-05 at 1 12 05 PM" src="https://user-images.githubusercontent.com/17363/183158633-35374331-8899-483e-9c3b-de8bc13fbf2f.png">

<img width="613" alt="Screen Shot 2022-08-05 at 1 16 22 PM" src="https://user-images.githubusercontent.com/17363/183158664-06dbfe53-67dc-43c9-b9f4-c3b1107616dc.png">

**Comment by jwalgran** [comment]
While testing my new Batch configuration in staging I learned that logs for Batch jobs end up in the general `/aws/batch/job` log group. 

According to [an issue in the terraform-aws-provider repo](https://github.com/hashicorp/terraform-provider-aws/issues/15560#issuecomment-710077627) changing the log stream for a job requires version 3.1 of the Terraform AWS provider. OS Hub currently used 2.7 (Terraform 0.11)

I found that through Log Insights it is possible to filter on a partial log stream name and the log streams includes the job name.

```
fields @timestamp, @message
| filter @logStream like /jobOpenSupplyHubStagingNotifications/
| sort @timestamp desc
| limit 20
```

This leads me to believe that don't strictly have to upgrade our Terraform and create dedicated log streams to view filtered notification log events, but this log group has a retention set to 1 month, and I anticipate wanting to see notification logs going back further than 1 month. It is easy to change the retention through the console, but:
- This will presumably change the retention for all batch job logs across all deployments.
- I do not know how this log group was created initially so I am concerned about manual edit being overwritten.

To make a good decision about how to proceed we need to know the level of effort required to upgrade us to a Terraform version that supports version 3.1 of the AWS provider.

**Comment by BryanQuigley** [comment]
I strongly recommend if we are doing a terraform upgrade to just make it all the way to a modern supported version (1.1.9) of both terraform and AWS (4.20..).  This also allows unblocks moving to github actions in the future. In previous projects there has also been some dropping of modules that are trivial to replace to streamline the upgrade (much easier to do if just a staging project). There are risks of staying on an intermittent step that we haven't really used in prod (except for upgrading).

If it was just upgrading one staging - I've done a number of upgrades from Terraform 0.11 (and older) to 1.1.9. (And AWS module 4.20ish). This can take under 4 hours to last [one](https://github.com/azavea/pfb-network-connectivity/pull/896/commits) took more like 12 hours (but now have solutions for some of the challenges there - the majority was actually the git issue).  The prod upgrade took another 3 hours.

The big complexity is that there are 3 environments here. Once terraform gets upgraded in one, if they all are based on that same terraform they are effectively locked from running terraform until they are manually upgraded as well.

To summarize:
If we can just do - one staging environment - I'd estimate 4-8 hours, possibly less if pairing
If we need to do all 3 environments - I'd estimate 10-16 hours. (6-10 hours initial, 2-3 per extra environment)

**Comment by jwalgran** [comment]
Thank you for the thoughtful and detailed response, Bryan.

We do have the option of updating just the single OS Hub staging environment which is being worked on in a separate `ogr/develop` branch. A complication is the fact that while it is called "staging" there is some valuable test data in the database and some partners are using the environment for testing. We do need to make care to not damage this environment.

@obrienad I will bring you in here and say that my recommendation is to go forward with updating our provisioning tools given that Bryan estimated it at 8 hours. We have time in the original estimate for this issue to cover it, and it gets us closer to accomplishing other architectural objectives like moving to Github actions.

**Comment by obrienad** [comment]
Thanks, Justin.

**Comment by jwalgran** [comment]
@BryanQuigley (CC @nanotubing) if you think that we can do the upgrade on our `ogr/develop` branch (just the OS Hub staging environment) without jeopardizing our deployment stability I would like to proceed with this upgrade. I think it makes sense to pair with Clause. When would you have time available to work with Claude on this?

**Comment by BryanQuigley** [comment]
Scheduled the upgrade for this coming Monday.

**Comment by jwalgran** [comment]
Excellent. Thank you. I appreciate you being able to jump in and help with this.
